sqlglot/dialects/spark.py

```diff
@@ -100,6 +100,8 @@ class Parser(Hive.Parser):
                 this=seq_get(args, 1),
                 unit=exp.var(seq_get(args, 0)),
             ),
+            "FLOAT": lambda args: exp.Cast(this=seq_get(args, 0), to=exp.DataType.build("float")),
+            "DOUBLE": lambda args: exp.Cast(this=seq_get(args, 0), to=exp.DataType.build("double")),
             "STRING": lambda args: exp.Cast(this=seq_get(args, 0), to=exp.DataType.build("string")),
```

Comment:
Should we dry these out? I'm thinking something like:

```python
...
def _parse_as_cast(to_type: str) -> exp.Expression:
    return lambda args: exp.Cast(this=seq_get(args, 0), to=exp.DataType.build(to_type))

...
            "FLOAT": _parse_as_cast("float"),
...
```

Reply:
Good idea! will just lift that suggestion, thanks!